scripts/salt-setup.sh

```diff
@@ -46,6 +46,53 @@ function apply_optional_states {
   fi
 }
 
+# This adds entities to the 'roles' key to be used during actual cluster install. Hence, this does not use the temporary config location.
+function add_single_role_for_cluster_salt {
+  local role=${1}
+  ${SALT_PATH}/bin/salt-call --local grains.append roles ${role} -l info --log-file=/tmp/salt-build-${saltenv}.log --log-file-level=debug
+}
+
+function add_prewarmed_roles {
+  if [ "${INCLUDE_FLUENT}" == "Yes" ]; then
+    # Note: This will need to be changed if making changes to versions etc in the prewarmed image.
+    local fluent_prewarmed="fluent_prewarmed_v1"
+    echo "Adding ${fluent_prewarmed} to the list of roles for the final image"
+    add_single_role_for_cluster_salt ${fluent_prewarmed}
+  fi
+
+  if  [ "${STACK_TYPE}" == "CDH" -a ! -z "${CLUSTERMANAGER_VERSION}" -a ! -z "${CLUSTERMANAGER_BASEURL}" -a ! -z "${CLUSTERMANAGER_GPGKEY}" -a ! -z "${STACK_VERSION}" -a ! -z "${STACK_BASEURL}" -a ! -z "${STACK_REPOID}" ]; then
+    local prewarmed="prewarmed_v1"
+    echo "Adding ${prewarmed} to the list of roles for the final image"
+    add_single_role_for_cluster_salt ${prewarmed}
+  fi
+}
+
+function create_archives_and_delete_files() {
+  tar -C /usr/lib64 -cvzf /usr/lib64/python3.6-archive.tar.gz python3.6
+  tar -C /usr/lib -cvzf /usr/lib/python3.6-archive.tar.gz python3.6
+  tar -C /usr/lib64 -cvzf /usr/lib64/python2.7-archive.tar.gz python2.7
+  tar -C /usr/lib -cvzf /usr/lib/python2.7-archive.tar.gz python2.7
+
+  # Salt (version as of this change: 3000.2) ends up taking a long time to load modules. vspere is an especially slow one taking 3 seconds.
+  # Salt does not seem to allow skipping module load ('disable_modules' only disables module usage, not module loading)
+  # So, deleting some modules which are not used, and tend to cause Exceptions / delays
+  find /opt/salt_3000.2 | grep modules | grep lxd | xargs rm
```

Comment:
these are a bit strange. why not use find for the whole?
`find /opt/salt_3000.2/lib/python3.6/site-packages/salt/modules/ -name "*lxd*"  -exec rm -f {} \;`

Reply:
No particular reason other than that's typically how I use find. Will change it over to to use find all the way.